crates/bevy_ui/src/accessibility.rs

```diff
@@ -0,0 +1,117 @@
+use bevy_a11y::{
+    accesskit::{kurbo::Rect, Node as AccessKitNode, Role},
+    AccessibilityNode,
+};
+use bevy_app::{App, CoreStage, Plugin};
+
+use bevy_ecs::{
+    prelude::Entity,
+    query::{Changed, Or, Without},
+    system::{Commands, Query},
+};
+use bevy_hierarchy::Children;
+
+use bevy_text::Text;
+use bevy_transform::prelude::GlobalTransform;
+use bevy_utils::default;
+
+use crate::{
+    prelude::{Button, Label},
+    Node, UiImage,
+};
+
+fn calc_name(texts: &Query<&Text>, children: &Children) -> Option<Box<str>> {
```

Comment:
It's not clear to me that we should be reading the text here, rather than using the `Name` component (and encouraging people to write to it).

Reply:
I agree in principle, but I'd like to see these sorts of ideas pushed forward in a general UI overhaul. Put in terms more familiar to those of us who work with accessibility, it's better when designed in from the start vs. bolted on later.

To be clear, I'm not critiquing bevy_ui for not being accessible from the start, particularly since bevy_ui is lacking quite a few things and accessibility is in good company. :) I'd just like to see us address these sorts of issues as part of a greater UI refactor of which accessibility is just one part. Without this groundwork, we wouldn't necessarily be able to meaningfully include accessibility as a design consideration when bevy_ui eventually gets more attention.